### PR TITLE
Do not be fooled by slices with the same name as the sliced element

### LINKED ISSUE #993: SUSHI instances don't like profiles w/ slicename matching path name(?)
See Zulip Thread: https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/error.20The.20element.20or.20path.20you.20referenced.20does.20not.20exist

After some investigation, I figured out that the problem occurs when the slicename matches the path name.  E.g., when `contact` is sliced with a slicename called `contact`.  Renaming the slicename to `contact2` fixes it!

Small reproducible sample: https://fshschool.org/FSHOnline/#/share/3egT4Lo

### PR DESCRIPTION
**Pull request by mint-thompson**
Fixes #993

When finding an element by path, it is permitted to use the name of a choice element slice (such as `valueString`) directly in a path. This is achieved by considering elements to match when the element id contains the appropriate path part and slice name (`value[x]:valueString`, for example). However, this should only be supported for choice elements, so do not consider an element to be a match when the name of the slice and the name of the element are the same.

Basically, without this change, the "remove slices" filter wouldn't remove the slice that had the same name as the element. Also, the bug only occurs when looking for a sufficiently deep descendant of the sliced element. What a thrill!

### CODE REVIEW

**Comment by cmoesel** [comment]
Good find, @mint-thompson.  That is wacky!  It looks good to me, but since it involves slicing logic, I'm going to go ahead and run a regression on it.

**Review by cmoesel — APPROVED** [review]
Thanks again, @mint-thompson.  This looks great.  The full regression passed:
```
No changes detected.
Total time: 3775 seconds
```

**Review by jafeltra — APPROVED** [review]
Tricky but makes sense!
